torch/_inductor/lowering.py

```diff
@@ -258,41 +258,74 @@ def in_namespace(op, namespace):
     return False
 
 
-def transform_args(args, broadcast, type_promotion_kind, convert_input_to_bool):
-    indices = [i for i, x in enumerate(args) if isinstance(x, TensorBox)]
-    if (type_promotion_kind or convert_input_to_bool) and indices:
+def transform_args(
+    args: List[Any],
+    kwargs: Dict[str, Any],
+    broadcast: bool,
+    type_promotion_kind: Optional[ELEMENTWISE_TYPE_PROMOTION_KIND],
+    convert_input_to_bool: bool,
+) -> Tuple[List[Any], Dict[str, Any]]:
+    args_indices = [i for i, x in enumerate(args) if isinstance(x, TensorBox)]
+    kwargs_indices = [k for k, v in kwargs.items() if isinstance(v, TensorBox)]
+    # check that there's something to transform
+    if not args_indices and not kwargs_indices:
+        return args, kwargs
+
+    if type_promotion_kind or convert_input_to_bool:
         if convert_input_to_bool:
             dtype = torch.bool
-        else:
+        # always true, but makes mypy happy when calling get_promoted_dtype
+        elif type_promotion_kind:
```

Comment:
type_promotion_kind can be None. I guess by always true you mean whenever convert_input_to_bool is set so is type_promotion_kind?

Reply:
Hmm. I'll reword this; it's trying to communicate that since the enclosing `if` statement triggers on `convert_input_to_bool or type_promotion_kind`, then if we didn't take the preceding `if` we _must_ be in this `elif`. I'm only doing an `elif` over an `else` because mypy was complaining about some code later in the function not checking whether `type_promotion_kind` was non-None.